lib/Runtime/Types/DeferredTypeHandler.cpp

```diff
@@ -24,7 +24,27 @@ namespace Js
 
         ScriptContext* scriptContext = instance->GetScriptContext();
         instance->EnsureSlots(0, typeHandler->GetSlotCapacity(), scriptContext, typeHandler);
-        typeHandler->SetInstanceTypeHandler(instance);
+
+        FunctionProxy * functionProxy = instance->GetFunctionProxy();
+        ScriptFunctionType * undeferredFunctionType = nullptr;
+        if (functionProxy)
+        {
+            undeferredFunctionType = functionProxy->GetUndeferredFunctionType();
+        }
+        if (undeferredFunctionType)
+        {
+            Assert(undeferredFunctionType->GetIsShared());
+            instance->ReplaceType(undeferredFunctionType);
```

Comment:
We should assert that undeferredFunctionType->GetTypeHandler() == typeHandler?

Reply:
I don't think we can do that. JavascriptLibrary::ConvertFunction calls this function with the shared SimpleTypeHandler from the library. That won't match the PathTypeHandler pointed to by undeferredFunctionType if we've done that conversion already.